hudi-spark/src/main/java/org/apache/hudi/OverwriteWithLatestAvroPayload.java

```
@@ -61,8 +60,16 @@ public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload
 
   @Override
   public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema) throws IOException {
+
+    GenericRecord genericRecord = (GenericRecord) getInsertValue(schema).get();
     // combining strategy here trivially ignores currentValue on disk and writes this record
-    return getInsertValue(schema);
+    Object deleteMarker = DataSourceUtils.getNestedFieldValOrNull(
+        genericRecord, "_hoodie_delete_marker");
+    if (deleteMarker != null && (boolean) deleteMarker) {
```

do we need a check for type and silently fail if its not a boolean?